src/transformStateWithClones.js

```diff
@@ -7,7 +7,31 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  return actions.reduce((acc, action, index) => {
+    const statePrevious = index === 0 ? { ...state } : { ...acc[index - 1] };
+
+    switch (action.type) {
+      case 'addProperties':
+        acc.push(Object.assign({}, statePrevious, action.extraData));
+        break;
+
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete statePrevious[key];
+        }
+        acc.push(Object.assign({}, statePrevious));
+        break;
+
+      case 'clear':
+        acc.push({});
+        break;
+
+      default:
+        acc.push(Object.assign({}, statePrevious, { ERROR: action.type }));
```

Comment:
Adding an `ERROR` key to the state copy is not described in the task requirements. It's important to stick to the defined task behavior. If an unknown action type is encountered, you should handle it appropriately without modifying the state.